frontend/components/SignupForm/SignupForm.tsx

```diff
@@ -1,98 +1,90 @@
 import React from 'react'
 import Link from 'next/link'
+import { useRouter } from 'next/router'
+import { useForm, ErrorMessage } from 'react-hook-form'
 import { trackCreateAccount } from '../../events/users'
-import validateAuth, { IErrors } from '../../lib/validateAuth'
 import { useCreateUserMutation } from '../../generated/graphql'
-import useFormValidation from '../../hooks/useFormValidation'
 import FormError from '../FormError'
 import Button from '../../elements/Button'
 import { brandBlue } from '../../utils'
-
-interface IFormValues {
-  handle: string
-  email: string
-  password: string
-}
-
-const initialState: IFormValues = {
-  handle: '',
-  email: '',
-  password: '',
-}
+import theme from '../../theme'
 
 const SignupForm: React.FC = () => {
-  const { handleChange, values, handleValidate, handleBlur, errors, setValues } = useFormValidation<
-    IFormValues,
-    IErrors
-  >(initialState, validateAuth)
-
-  const [createUser, { loading, error }] = useCreateUserMutation({
-    onCompleted: () => {
-      trackCreateAccount()
-      setValues(initialState)
-    },
+  const router = useRouter()
+  const { handleSubmit, register, errors } = useForm({
+    mode: 'onBlur',
   })
 
-  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault()
-    handleValidate(e)
+  const fieldErrorName = Object.keys(errors)[0] || ''
+
+  const [createUser, { loading, error }] = useCreateUserMutation()
+
+  const onSubmit = (data: any) => {
     if (!loading && Object.keys(errors).length === 0) {
       createUser({
         variables: {
-          handle: values.handle,
-          email: values.email,
-          password: values.password,
+          handle: data.handle,
+          email: data.email,
+          password: data.password,
         },
       })
+      trackCreateAccount()
+      router.push({
+        pathname: '/dashboard/my-feed',
+      })
     }
   }
 
   return (
-    <form onSubmit={handleSubmit}>
+    <form onSubmit={handleSubmit(onSubmit)}>
       <fieldset disabled={loading} aria-busy={loading}>
         <h2>Sign up for an account</h2>
         {error && <FormError error={error} />}
         <label htmlFor="handle">
-          Username
-          <p>{errors?.handle}</p>
+          Display Name
```

Comment:
Can we translate all text that appears in the form? `/frontend/public/static/locales/en/settings.json` has translations for all the settings form values, like labels and form error messages, if you want to use that as a guide for structuring a new translation file. We could create something like `account.json` perhaps that could house text for both the login and signup forms (I'm assuming some text values will be the same across the two forms)

Reply:
As discussed in my audio message, for this PR I was just trying to make a bunch of these quick fixes for things that are really broken aesthetically or that were major blockers to the UX or even completing a flow (like the routing issue). I propose to handle this in another PR 🙂

Reply:
Sounds good!!